packages/flame/lib/src/geometry/shape_component.dart

```diff
@@ -16,8 +16,14 @@ abstract class ShapeComponent extends PositionComponent with HasPaint {
     super.children,
     super.priority,
     Paint? paint,
+    List<Paint>? paintLayers,
   }) {
-    this.paint = paint ?? this.paint;
+    if (paint != null) {
+      this.paint = paint;
+    }
+    if (paintLayers != null) {
+      this.paintLayers = paintLayers;
+    }
```

Comment:
```suggestion
    this.paint = paint ?? this.paint;
    this.paintLayers = paintLayers ?? this.paintLayers;
```

Reply:
I changed this on the last commit as to me it reads that if paintLayers in the constructor is null then get this.paintLayers. If the getter is called in `HasPaint` then it allocates memory and creates the `List`. We're trying to avoid this if paintLayers are not used?

Reply:
[left unchanged after agreed in discord this.paintLayers should not be read if constructor paintLayers null to prevent get in HasPaint allocating a List unnecessarily]